### PR TITLE
feat: drop Node.js 16

### PR DESCRIPTION
**Pull request by balazsorban44**
### What?

BREAKING CHANGE: Bump the minimum required Node.js version.

### Why?

Node.js 16 has reached end-of-life in September.

Bumped to `18.18.2` since it contained some security-related patches: https://nodejs.org/en/blog/vulnerability/october-2023-security-releases

### How?

Bumped `engines` where needed, upgraded the workflows.

This will allow us to remove quite a few polyfills, I'll open separate PRs for those.

### CODE REVIEW

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 79b314d185dd75af925de2ab3b19ab83d16d2ecf

`pnpm test-dev test/e2e/app-dir/next-image/next-image-proxy.test.ts` 

<details>
<summary>Expand output</summary>



  ● Test suite failed to run

    Failed to proxy: read ECONNRESET

      41 |
      42 |       proxy.on('error', (err) => {
    > 43 |         throw new Error('Failed to proxy: ' + err.message)
         |               ^
      44 |       })
      45 |
      46 |       await new Promise<void>((resolve) => {

      at ProxyServer.<anonymous> (e2e/app-dir/next-image/next-image-proxy.test.ts:43:15)
      at ProxyServer.emit (../node_modules/.pnpm/eventemitter3@4.0.7/node_modules/eventemitter3/index.js:210:27)
      at ClientRequest.proxyError (../node_modules/.pnpm/http-proxy@1.18.1/node_modules/http-proxy/lib/http-proxy/passes/web-incoming.js:165:18)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

File: docs/04-architecture/supported-browsers.mdx [hunk 1]
```diff
@@ -57,10 +57,6 @@ Next.js allows you to use the latest JavaScript features out of the box. In addi
 - [Class Fields](https://github.com/tc39/proposal-class-fields) and [Static Properties](https://github.com/tc39/proposal-static-class-features) (part of stage 3 proposal)
 - and more!
 
-### Server-Side Polyfills
```
Comment (balazsorban44) [inline docs/04-architecture/supported-browsers.mdx:60]: `fetch` is not polyfilled anymore (not needed), but I think this section could be its own page anyway.

**Comment by ijjk** [comment]
# Stats from current PR

<details open>
<summary><strong>Default Build</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | vercel/next.js chore/bump-node | Change |
| - | - | - | - |
| buildDuration | 10.4s | 10.6s | ⚠️ +186ms |
| buildDurationCached | 6.3s | 6.2s | N/A |
| nodeModulesSize | 172 MB | 172 MB | N/A |
| nextStartRea..uration (ms) | 530ms | 534ms | N/A |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js chore/bump-node | Change |
| - | - | - | - |
| 199-HASH.js gzip | 27.5 kB | 27.5 kB | N/A |
| 3f784ff6-HASH.js gzip | 50.9 kB | 50.9 kB | ✓ |
| 99.HASH.js gzip | 182 B | 182 B | ✓ |
| framework-HASH.js gzip | 45.3 kB | 45.3 kB | ✓ |
| main-app-HASH.js gzip | 254 B | 251 B | N/A |
| main-HASH.js gzip | 32.9 kB | 32.9 kB | N/A |
| webpack-HASH.js gzip | 1.75 kB | 1.75 kB | N/A |
| Overall change | 96.4 kB | 96.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js chore/bump-node | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js chore/bump-node | Change |
| - | - | - | - |
| _app-HASH.js gzip | 206 B | 205 B | N/A |
| _error-HASH.js gzip | 182 B | 180 B | N/A |
| amp-HASH.js gzip | 506 B | 505 B | N/A |
| css-HASH.js gzip | 322 B | 323 B | N/A |
| dynamic-HASH.js gzip | 2.57 kB | 2.57 kB | N/A |
| edge-ssr-HASH.js gzip | 260 B | 259 B | N/A |
| head-HASH.js gzip | 350 B | 350 B | ✓ |
| hooks-HASH.js gzip | 369 B | 369 B | ✓ |
| image-HASH.js gzip | 4.35 kB | 4.35 kB | N/A |
| index-HASH.js gzip | 256 B | 256 B | ✓ |
| link-HASH.js gzip | 2.64 kB | 2.63 kB | N/A |
| routerDirect..HASH.js gzip | 312 B | 311 B | N/A |
| script-HASH.js gzip | 385 B | 384 B | N/A |
| withRouter-HASH.js gzip | 307 B | 308 B | N/A |
| 1afbb74e6ecf..834.css gzip | 106 B | 106 B | ✓ |
| Overall change | 1.08 kB | 1.08 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js chore/bump-node | Change |
| - | - | - | - |
| _buildManifest.js gzip | 485 B | 482 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js chore/bump-node | Change |
| - | - | - | - |
| index.html gzip | 529 B | 528 B | N/A |
| link.html gzip | 542 B | 541 B | N/A |
| withRouter.html gzip | 525 B | 522 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Edge SSR bundle Size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js chore/bump-node | Change |
| - | - | - | - |
| edge-ssr.js gzip | 93.7 kB | 93.7 kB | N/A |
| page.js gzip | 152 kB | 152 kB | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Middleware size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js chore/bump-node | Change |
| - | - | - | - |
| middleware-b..fest.js gzip | 626 B | 621 B | N/A |
| middleware-r..fest.js gzip | 150 B | 151 B | N/A |
| middleware.js gzip | 22.5 kB | 22.5 kB | N/A |
| edge-runtime..pack.js gzip | 1.92 kB | 1.92 kB | ✓ |
| Overall change | 1.92 kB | 1.92 kB | ✓ |

</details>

</details>
Commit: 73e2c7ffc24f206c12ad98e5df5f84c941a3c90d

File: .github/workflows/build_and_deploy.yml [hunk 2]
```diff
@@ -8,8 +8,8 @@ on:
 env:
   NAPI_CLI_VERSION: 2.16.2
   TURBO_VERSION: 1.10.9
-  NODE_MAINTENANCE_VERSION: 16
-  NODE_LTS_VERSION: 18
+  NODE_MAINTENANCE_VERSION: 18.18.2
```
Comment (styfle) [inline .github/workflows/build_and_deploy.yml:11]: Can we test agains 18 so it automatically tests minor and patch releases instead of pinning?

Comment (balazsorban44) [reply .github/workflows/build_and_deploy.yml:11]: I did so first, but changed in https://github.com/vercel/next.js/pull/56896/commits/6c00b6942b8836a803aed29970a85534b14f010c as <18.18.2 was being installed, which caused issues.

See https://github.com/vercel/next.js/actions/runs/6536411624/job/17748103891?pr=56896#step:28:94

Comment (balazsorban44) [reply .github/workflows/build_and_deploy.yml]: Actually, that value seems to be unused, it was just being copied over from other similar workflows.

**Review by ijjk — APPROVED** [review]

**Comment by hanayashiki** [comment]
@ljharb

**Comment by ljharb** [comment]
@hanayashiki yes? Why am i being tagged here?

**Pull request merged**